idunn/datasources/pages_jaunes.py

```diff
@@ -58,21 +75,87 @@ def get_places_bbox(self, raw_categories, bbox, size=10, query=""):
         raw_places = result.get("hits", {}).get("hits", [])
         return [PjPOI(p["_source"]) for p in raw_places]
 
-    def get_place(self, id):
-        internal_id = id.replace(f"{self.PLACE_ID_NAMESPACE}:", "", 1)
-
+    def get_place(self, poi_id):
         es_places = self.es.search(
             index=self.es_index,
-            body={"query": {"bool": {"filter": {"term": {"_id": internal_id}}}}},
+            body={"query": {"bool": {"filter": {"term": {"_id": self.internal_id(poi_id)}}}}},
             ignore_unavailable=True,
         )
 
         es_place = es_places.get("hits", {}).get("hits", [])
         if len(es_place) == 0:
-            raise HTTPException(status_code=404, detail=f"place {id} not found")
+            raise HTTPException(status_code=404, detail=f"place {poi_id} not found")
         if len(es_place) > 1:
-            logger.warning("Got multiple places with id %s", id)
+            logger.warning("Got multiple places with id %s", poi_id)
         return PjPOI(es_place[0]["_source"])
 
 
-pj_source = PjSource()
+class PjAuthSession(AuthSession):
+    def get_authorization_url(self):
+        return "https://api.pagesjaunes.fr/oauth/client_credential/accesstoken"
+
+    def get_authorization_params(self):
+        return {
+            "grant_type": "client_credentials",
+            "client_id": settings.get("PJ_API_ID"),
+            "client_secret": settings.get("PJ_API_SECRET"),
+        }
+
+
+class ApiPjSource(PjSource):
+    PJ_RESULT_MAX_SIZE = 30
+    PJ_INFO_API_URL = "https://api.pagesjaunes.fr/v1/pros"
+    PJ_FIND_API_URL = "https://api.pagesjaunes.fr/v1/pros/search"
+
+    def __init__(self):
+        self.enabled = True
+        self.session = PjAuthSession()
+
+    @staticmethod
+    def format_where(bbox):
+        left, bot, right, top = bbox
+        return f"gZ{left},{top},{right},{bot}"
+
+    def get_from_params(self, url, params=None) -> PjApiPOI:
+        res = self.session.get(url, params=params)
+        res.raise_for_status()
+        return res.json()
+
+    def get_places_from_url(self, url, params=None, size=10):
+        res = pj_find.Response(**self.get_from_params(url, params))
+        pois = [PjApiPOI(listing) for listing in res.search_results.listings[:size] or []]
+
+        if (
+            len(pois) < size
+            and res.context
+            and res.context.pages
+            and res.context.pages.next_page_url
+        ):
+            pois += self.get_places_from_url(
+                res.context.pages.next_page_url.replace("/pros/find", "/pros/search"),
+                size=size - len(pois),
+            )
+
+        return pois
+
+    def get_places_bbox(self, raw_categories, bbox, size=10, query="") -> List[PjApiPOI]:
+        query_params = {
+            "what": " ".join(raw_categories),
+            "where": self.format_where(bbox),
+            "max": min(self.PJ_RESULT_MAX_SIZE, size),
```

Comment:
Note: before merging we need to decide whether we should keep this check, or whether successive calls to fetch several pages are acceptable

Reply:
This max is the query parameter sent to the API, although current implementation will apply the successive calls in `get_places_from_url` by opening recursively `next_page_url`